sdk/storage/azure-storage-blob-nio/src/main/java/com/azure/storage/blob/nio/AzureFileSystemProvider.java

```diff
@@ -1152,30 +1209,42 @@ void closeFileSystem(String fileSystemName) {
         this.openFileSystems.remove(fileSystemName);
     }
 
-    private String extractAccountEndpoint(URI uri) {
+    private String extractAccountEndpointOrGetDefault(URI uri) {
         if (!uri.getScheme().equals(this.getScheme())) {
             throw LoggingUtility.logError(ClientLoggerHolder.LOGGER, new IllegalArgumentException(
                 "URI scheme does not match this provider"));
         }
-        if (CoreUtils.isNullOrEmpty(uri.getQuery())) {
+        if (CoreUtils.isNullOrEmpty(uri.getQuery()) && CoreUtils.isNullOrEmpty(DEFAULT_ENDPOINT)) {
             throw LoggingUtility.logError(ClientLoggerHolder.LOGGER,
                 new IllegalArgumentException("URI does not contain a query component. FileSystems require a URI of "
                     + "the format \"azb://?endpoint=<account_endpoint>\"."));
         }
 
-        String endpoint = Flux.fromArray(uri.getQuery().split("&"))
-                .filter(s -> s.startsWith(ENDPOINT_QUERY_KEY + "="))
-                .switchIfEmpty(Mono.defer(() -> Mono.error(LoggingUtility.logError(ClientLoggerHolder.LOGGER,
-                    new IllegalArgumentException("URI does not contain an \"" + ENDPOINT_QUERY_KEY + "=\" parameter. "
-                        + "FileSystems require a URI of the format \"azb://?endpoint=<endpoint>\"")))))
-                .map(s -> s.substring(ENDPOINT_QUERY_KEY.length() + 1)) // Trim the query key and =
-                .blockLast();
+        String endpoint = Arrays.stream((uri.getQuery() != null ? uri.getQuery() : "").split("&"))
+            .filter(s -> s.startsWith(ENDPOINT_QUERY_KEY + "="))
+            .map(s -> s.substring(ENDPOINT_QUERY_KEY.length() + 1)) // Trim the query key and =
+            .findFirst().orElse(DEFAULT_ENDPOINT);
 
         if (CoreUtils.isNullOrEmpty(endpoint)) {
             throw LoggingUtility.logError(ClientLoggerHolder.LOGGER,
-                new IllegalArgumentException("No account endpoint provided in URI query."));
+                new IllegalArgumentException("No account endpoint provided in URI query and no default configured."));
         }
 
         return endpoint;
     }
+
+    static AzureFileSystemConfig readEnvironmentConfiguration() {
+        com.azure.core.util.Configuration config = com.azure.core.util.Configuration.getGlobalConfiguration();
+        return new AzureFileSystemConfig(config);
+    }
+
+    static String readEnvironmentEndpoint() {
+        return com.azure.core.util.Configuration.getGlobalConfiguration().get(
```

Comment:
Perhaps this should have checks on whether the global configs have already been read? Or if the endpoint itself has already been read? Not sure if getGlobalConfiguration caches or what the perf is like

Reply:
EnvironmentConfiguration looks like it caches. I still have some more testing to do around default endpoints, though.